Update MauticInstaller to allow plugins & themes

nickveenhof opened the pull request:
Update MauticInstaller to allow plugins & themes with custom directory names - this has been verified (composer create-project mautic/recommended-project:4.x-dev mauticcomposerproject --no-interaction) and can be committed!

I know the suggestion is to use installer-name, but there was a composer library https://github.com/mautic/composer-plugin that already had the convention of install-directory-name so I wanted to keep compatibility with that.

niksamokhvalov reviewed (changes requested):
Update tests, please.

nickveenhof commented:
Tests added - please re-review. Thanks!

niksamokhvalov commented:
https://github.com/composer/installers/pull/484/checks?check_run_id=2435477260
https://github.com/composer/installers/pull/484/checks?check_run_id=2435477049

These checks is failing. Fix it, please.

nickveenhof commented:
Fixed the failing tests, was only not able to reproduce the PHP 5.3 test locally but I think I fixed it due to PHP 7 array syntaxes. Tests still work and phpstan is now also happy
